R/quick_functions.R

```diff
@@ -67,3 +69,53 @@ viz_kpi_finyear <- function(df) {
 }
 
 
+
+# general -----------------------------------------------------------------
+
+## simplify_fy ----
+
+#' Convert long-form financial year (20XX/YY) into short-form (XXYY)
+#'
+#' @param financial_year Character string object/vector of financial year in format "20XX/YY"
+#'
+#' @return Character string object/vector of financial year in format "XXYY"
+#' @export
+#'
+#' @examples
+#'
+#' simplify_fy("2023/24")
+#' [1] "2324"
+#'
+simplify_fy <- function(financial_year) {
+
+  # checks
+  stopifnot("financial_year input is not character object" = is.character(financial_year) == TRUE)
+  stopifnot("financial_year input is not in format XXXX/YY" = nchar(financial_year) == 7)
+  stopifnot("financial_year input is not in format XXXX/YY" = substr(financial_year, 5, 5) == "/")
+  stopifnot("financial_year input years are not consecutive" = (as.numeric(substr(financial_year, 6, 7)) - as.numeric(substr(financial_year, 3, 4))) == 1)
+
+
+  paste0(substr(financial_year, 3, 4), substr(financial_year, 6, 7))
+}
+
+## format_excel_hyperlink ----
+
+#' Turn URL into hyperlink for Excel outputs
+#'
+#' @param name Character string text displayed for link
+#' @param url Character string URL
+#'
+#' @return 1x1 dataframe containing hyperlink ready for writing with openxlsx
+#' @export
+#'
+#' @examples
+#' format_excel_hyperlink("Search engine", "www.google.com")
+#'
+format_excel_hyperlink <- function(name, url) {
```

Comment:
Not sure if it will do the same, but do you know of writeFormula() in openxlsx?

I use it for the excel we upload with the publication:
writeFormula(wb, sheet = 1,
             x = '=HYPERLINK("https://www.gov.scot/collections/scottish-index-of-multiple-deprivation-2020/",
             "1 Scottish Index of Multiple Deprivation 2020 (Quintiles)")',
             startRow = 33, startCol = 1)

Reply:
Ah is the bit after 'x' the text that displays?

Reply:
x = '=HYPERLINK("https://www.gov.scot/collections/scottish-index-of-multiple-deprivation-2020/",
"1 Scottish Index of Multiple Deprivation 2020 (Quintiles)")'

The whole thing after x in the ' ' is the formula in the cell, containing the hyperlink in the first " " within the brackets of =HYPERLINK(), and then the text as displayed in the second " ".

https://www.rdocumentation.org/packages/openxlsx/versions/4.2.7.1/topics/writeFormula